ports/esp32/mphalport.c

```diff
@@ -78,6 +84,24 @@ void check_esp_err(esp_err_t code) {
         o_str->data = (const byte *)esp_err_to_name(code); // esp_err_to_name ret's ptr to const str
         o_str->len = strlen((char *)o_str->data);
         o_str->hash = qstr_compute_hash(o_str->data, o_str->len);
+
+        #if MICROPY_ERROR_REPORTING > MICROPY_ERROR_REPORTING_NORMAL
+        mp_printf(MP_PYTHON_PRINTER, "Exception in function '%s' at line %d in file '%s'\n", func, line, file);
+        char err_msg[64];
+        esp_err_to_name_r(code, err_msg, sizeof(err_msg));
+        char str[256];
+        snprintf(str, sizeof(str), MP_ERROR_TEXT("0x%04X, in function '%s' at line %d in file '%s'"), code, func, line, file);
```

Comment:
snprintf won't work with `MP_ERROR_TEXT`.  Just remove the `MP_ERROR_TEXT` macro.